woocommerce/woocommerce-android
# Issue/7053 track linked products

This PR partly addresses #7053 by adding missing tracking of linked products.

The reason for the missed tracking was that the tracking was being done on the grouped product list screen, which is bypassed when there are no linked products yet. I checked on iOS and it shows the grouped product list with a "No products yet" empty view in this situation, so I chose to match iOS.

To test:

* Open product detail for a product that has no linked products
* Tap "Add more details"
* Tap "Linked products"
* Tap "Add products"
* Verify the grouped product list is shown with an empty view
* Tap "Add product" and verify that `connected_products_list` is tracked with `action: add_tapped`
* Select a product and verify that `connected_products_list` is tracked with `action: added


**Android empty linked product list:**

![upsells](https://user-images.githubusercontent.com/3903757/183496780-fd8da072-56a6-4383-a9c0-84247ca67d3f.png)

**iOS empty linked product list:**

![iOS](https://user-images.githubusercontent.com/3903757/183496862-3df85150-5a0c-496f-9220-761313bef3c3.png)
